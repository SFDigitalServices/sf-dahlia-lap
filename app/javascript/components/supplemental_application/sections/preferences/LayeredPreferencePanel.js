import React from 'react'

import FormGrid from 'components/molecules/FormGrid'

import { Comment, FormItem } from './utils'

export const LayeredPreferencePanel = ({ preference }) => {
  return (
    <>
      <FormGrid.Row expand={false}>
        <FormItem label='Preference Name'>
          <div className='text-value'>{'Veteran status'}</div>
        </FormItem>
        <FormItem label='HH Member on Proof'>
          <div className='text-value'>{preference.person_who_claimed_name}</div>
        </FormItem>
        <FormItem label='Type of Proof'>
          <div className='text-value'>
            {preference.veteran_type_of_proof ? preference.veteran_type_of_proof : 'None'}
          </div>
        </FormItem>
        <FormItem label='Status'>
          <div className='text-value'>{preference.post_lottery_validation}</div>
        </FormItem>
      </FormGrid.Row>
      <FormGrid.Row expand={false}>
        <div className='form-group'>
          <div className='form-grid_item large-12 column padding-bottom'>
            <Comment>{'Veteran status already confirmed by MOHCD.'}</Comment>
          </div>
        </div>
      </FormGrid.Row>
    </>
  )
}

export default LayeredPreferencePanel
