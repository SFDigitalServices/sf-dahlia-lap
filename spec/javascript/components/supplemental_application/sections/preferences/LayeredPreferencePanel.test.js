import React from 'react'

import { render } from '@testing-library/react'

import LayeredPreferencePanel from 'components/supplemental_application/sections/preferences/LayeredPreferencePanel'

const preference = {
  person_who_claimed_name: 'John Doe',
  veteran_type_of_proof: 'DD Form 214',
  receives_preference: true,
  post_lottery_validation: 'Unconfirmed'
}

const preference_invalid = {
  person_who_claimed_name: 'John Doe',
  veteran_type_of_proof: 'DD Form 214',
  receives_preference: true,
  post_lottery_validation: 'Invalid'
}

describe('LayeredPreferencePanel', () => {
  test('should render successfully', () => {
    const { asFragment } = render(<LayeredPreferencePanel preference={preference} />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('should handle when preference is invalid', () => {
    const { asFragment } = render(<LayeredPreferencePanel preference={preference_invalid} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
