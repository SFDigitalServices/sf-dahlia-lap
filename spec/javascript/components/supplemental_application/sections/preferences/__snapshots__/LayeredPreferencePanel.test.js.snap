// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayeredPreferencePanel should render successfully 1`] = `
<DocumentFragment>
  <div
    class="form-grid row margin-left-none margin-right-none"
  >
    <div
      class="form-grid_item column small-6"
    >
      <div
        class="form-group"
      >
        <label
          for=""
        >
          Preference Name
        </label>
        <div
          class="text-value"
        >
          Veteran status
        </div>
      </div>
    </div>
    <div
      class="form-grid_item column small-6"
    >
      <div
        class="form-group"
      >
        <label
          for=""
        >
          HH Member on Proof
        </label>
        <div
          class="text-value"
        >
          John Doe
        </div>
      </div>
    </div>
    <div
      class="form-grid_item column small-6"
    >
      <div
        class="form-group"
      >
        <label
          for=""
        >
          Type of Proof
        </label>
        <div
          class="text-value"
        >
          DD Form 214
        </div>
      </div>
    </div>
    <div
      class="form-grid_item column small-6"
    >
      <div
        class="form-group"
      >
        <label
          for=""
        >
          Status
        </label>
        <div
          class="text-value"
        >
          Unconfirmed
        </div>
      </div>
    </div>
  </div>
  <div
    class="form-grid row margin-left-none margin-right-none"
  >
    <div
      class="form-group"
    >
      <div
        class="form-grid_item large-12 column padding-bottom"
      >
        <p
          class="t-base c-steel"
        >
          Veteran status already confirmed by MOHCD.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RentBurdenedPanel should render successfully 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form-grid row margin-left-none margin-right-none"
    >
      <div
        class="form-grid_item column small-6"
      >
        <div
          class="form-group"
        >
          <label
            for=""
          >
            Preference Name
          </label>
          <div
            class="text-value"
          >
            Rent Burdened Housing Preference
          </div>
        </div>
      </div>
      <div
        class="form-grid_item column small-6"
      >
        <div
          class="form-group"
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="form-total_monthly_rent"
              id="label-total_monthly_rent"
            >
              Total Household Monthly Rent
            </label>
            <input
              class=""
              id="form-total_monthly_rent"
              name="total_monthly_rent"
              placeholder="Enter Amount"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="form-grid_item column small-6"
      >
        <div
          class="form-group"
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="form-preferences.1.post_lottery_validation"
              id="label-preferences-1.post_lottery_validation"
            >
              Status
            </label>
            <select
              class="preference-status-select"
              id="form-preferences.1.post_lottery_validation"
              name="preferences.1.post_lottery_validation"
            >
              <option
                disabled=""
                value=""
              >
                Select One...
              </option>
              <option
                value="Unconfirmed"
              >
                Unconfirmed
              </option>
              <option
                value="Confirmed"
              >
                Confirmed
              </option>
              <option
                value="Invalid"
              >
                Invalid
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
